### DISCUSSION TITLE
Textual web app

### DISCUSSION THREAD

**Discussion by jrezai** [Textualize/textual#455] [Q&A] [answered]
On the site [textualize.io](https://textualize.io/), it mentions running Textual apps in a web browser. Can you provide more information on that?

How will that work? When the web feature is released, will Textual apps be able to run from a web browser only through your website ([textualize.io](https://textualize.io/)) via a paid subscription? Or will a Textual app be able to run from a self-hosted server? Will the web app feature be proprietary or open-source?

Thanks.

**Comment by willmcgugan** [comment, marked as answer]
Initially it will a self-hosted service. You run your textual app with switch that turns it in to a server of sorts. You will get back a URL on textualize.io that you can distribute.

There will be a very generous first tier, making it essentially free for most people. Additional features / scale will require a paid subscription. The web-app side will be propriety.
